pandas/core/arrays/arrow/array.py

```diff
@@ -81,10 +81,10 @@
     }
 
     ARROW_LOGICAL_FUNCS = {
-        "and": pc.and_kleene,
-        "rand": lambda x, y: pc.and_kleene(y, x),
-        "or": pc.or_kleene,
-        "ror": lambda x, y: pc.or_kleene(y, x),
+        "and_": pc.and_kleene,
+        "rand_": lambda x, y: pc.and_kleene(y, x),
+        "or_": pc.or_kleene,
+        "ror_": lambda x, y: pc.or_kleene(y, x),
         "xor": pc.xor,
```

Comment:
I think(?) the first four here should have underscores. These were raising in an ASV.

Reply:
for example, this was raising:

```
import pandas as pd

s1 = pd.Series([True], dtype="boolean[pyarrow]")
s2 = pd.Series([False], dtype="boolean[pyarrow]")

# raises: KeyError: 'or_'
s1 | s2  
```

Reply:
Good find! Could you move into a separate pr and add tests?

Reply:
sure, will do

Reply:
opened #51688 for this